examples/auto-relay/listener.js

```diff
@@ -0,0 +1,50 @@
+'use strict'
+
+const Libp2p = require('libp2p')
+const Websockets = require('libp2p-websockets')
+const { NOISE } = require('libp2p-noise')
+const MPLEX = require('libp2p-mplex')
+
+async function main () {
+  const relayAddr = process.argv[2]
+  if (!relayAddr) {
+    throw new Error('the relay address needs to be specified as a parameter')
+  }
+
+  const node = await Libp2p.create({
+    modules: {
+      transport: [Websockets],
+      connEncryption: [NOISE],
+      streamMuxer: [MPLEX]
+    },
+    config: {
+      relay: {
+        enabled: true,
+        autoRelay: {
+          enabled: true,
+          maxListeners: 2
+        }
+      }
+    }
+  })
+
+  await node.start()
+  console.log(`Node started with id ${node.peerId.toB58String()}`)
+
+  const conn = await node.dial(relayAddr)
+
+  // Wait for connection and relay to be bind for the example purpose
+  await new Promise((resolve) => {
+    node.peerStore.on('change:multiaddrs', ({ peerId }) => {
+      // Updated self multiaddrs?
+      if (peerId.equals(node.peerId)) {
+        resolve()
```

Comment:
Maybe get rid of the promise logic and just put the console logs here? This event listener never gets cleaned up and can throw an error if our addresses change again and call `resolve` on the already resolved promise.

Reply:
yes, that makes a lot of sense! thanks